### ISSUE TITLE
[java] UnusedPrivateMethod: False positive with raw type for generic method

### ISSUE THREAD

**Issue by Pankraz76** [pmd/pmd#6028]
**Affects PMD Version:** PMD 7.14.0

**Rule:** [UnusedPrivateMethod](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:**

**Code Sample demonstrating the issue:**

```java
import java.util.Collection;

public final class addAllImpl {
  private static abstract class Multiset<K> extends Collection<K> { }
  static <E> boolean addAllImpl(
      Multiset<E> self, Collection<?> elements) {
    return addAllImpl(self, (Multiset) elements);
  }

  private static <E> boolean addAllImpl(
      Multiset<E> self, Multiset<E> elements) {
  }
}
```

**Expected outcome:**

PMD reports a violation at line 10, but that's wrong. That's a false positive.

**Running PMD through:** maven-pmd-plugin 3.27.0

**Comment by UncleOwen** [comment]
Your issues are extremely difficult to read. Please

- Shorten your code samples so that they clearly demonstrate the issue.
- Use the provided template as it is intended. You managed to fill in every single part wrong.

**Comment by Pankraz76** [comment]
> * You managed to fill in every single part wrong.

its just the template, right?


updated the test data

**Comment by UncleOwen** [comment]
If you want unreadable issues, it's "just" the template. Otherwise, it's a helpful tool to structure information.

**Comment by Pankraz76** [comment]
yes you right, sorry. 

Actually provided key information which is the recipe name and the test code thats all needed to reproduce. 

Anything else is considered obsolete from my point of view.

**Comment by zbynek** [comment]
The testcase can be simplified by removing the imports:

```java
import java.util.Collection;

public final class addAllImpl {
  private static abstract class Multiset<K> extends Collection<K> { }
  static <E> boolean addAllImpl(
      Multiset<E> self, Collection<?> elements) {
    return addAllImpl(self, (Multiset) elements);
  }

  private static <E> boolean addAllImpl(
      Multiset<E> self, Multiset<E> elements) {
  }
}
```

Looks like overload resolution is not working well in a particular case involving generics.

**Comment by oowekyala** [comment]
Yes this looks like it's related to raw types. Maybe related to #5097
